### PR TITLE
Move reference content to new REFERENCE.md

### PR DESCRIPTION
**Pull request by izgeri**
### What does this PR do?
In the Puppet [docs on writing modules](https://puppet.com/docs/puppet/5.5/bgtm.html#documenting-your-module), they advise to put the reference content in a REFERENCE.md file. This commit moves our existing reference into this (new)
file; the reference still needs to be reviewed and improved in follow-oncommits.

### What ticket does this PR close?
n/a

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

**Comment by izgeri** [comment]
Filed a follow up issue in #166 to migrate the REFERENCE to be generated using `puppet strings` based on code comments.

File: REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+API key for a user or host. Must be `Sensitive` if supported.
+
+##### `cert_file`
+File path to X509 certificate of the root CA of Conjur, PEM formatted. This file is read
+from the **Puppet server**. Takes precedence over `ssl_certificate`.
+
+##### `ssl_certificate`
+Content of the X509 certificate of the root CA of Conjur, PEM formatted.
+When using Puppet's `file` function, the path to the cert must be absolute.
+Overwritten by the contents read from `cert_file` when it is present.
```
Comment (sgnn7) [inline REFERENCE.md:63]: I see a _lot_ of references to cert file here. you may need a rebase and manual fixes on this for that

File: REFERENCE.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+#### Parameters
+
+##### `account`
+Conjur account authority name. Optional for v4, required for v5.
+
+##### `appliance_url`
+A Conjur endpoint (with trailing `/api/` for v4).
+
+##### `authn_login`
+User username or host name (prefixed with `host/`).
```
Comment (sgnn7) [inline REFERENCE.md:51]:
```suggestion
User's username or host name (prefixed with `host/`).
```

File: REFERENCE.md (hunk 1 above)
Comment (izgeri) [reply REFERENCE.md:63]: yup, gonna need a rebase. will work on that now.

**Comment by izgeri** [comment]
@sgnn7 updated.

File: REFERENCE.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  host_factory_token => Sensitive('f9yykd2r0dajz398rh32xz2fxp1tws1qq2baw4112n4am9x3ncqbk3')
+}
+
+# using an API key
+class { 'conjur':
+  account         => 'mycompany',
+  appliance_url   => 'https://conjur.mycompany.com/',
+  authn_login     => 'host/redis001',
+  authn_api_key   => Sensitive('f9yykd2r0dajz398rh32xz2fxp1tws1qq2baw4112n4am9x3ncqbk3'),
+  ssl_certificate => file('/abslute/path/to/conjur-ca.pem'),
```
Comment (sgnn7) [inline REFERENCE.md:97]:
```suggestion
  ssl_certificate => file('/absolute/path/to/conjur-ca.pem'),
```

**Pull request merged**